third_party/stdlib/collections.py

```diff
@@ -258,37 +258,43 @@ def viewitems(self):
 ### namedtuple
 ################################################################################
 
-_class_template = '''\
-class {typename}(tuple):
+class NamedTuple(tuple):
     '{typename}({arg_list})'
 
     __slots__ = ()
 
-    _fields = {field_names!r}
+    _fields = ()
 
-    def __new__(_cls, {arg_list}):
+    def __new__(_cls, *args, **kwds):
         'Create new instance of {typename}({arg_list})'
-        return _tuple.__new__(_cls, ({arg_list}))
+        try:
+            kwargs = tuple(map(kwds.pop, list(_cls._fields[len(args):])))
+        except KeyError as e:
+            raise TypeError('Unexpected keyword argument %r' % e.message)
+        return tuple.__new__(_cls, args + kwargs)
 
     @classmethod
     def _make(cls, iterable, new=tuple.__new__, len=len):
         'Make a new {typename} object from a sequence or iterable'
         result = new(cls, iterable)
-        if len(result) != {num_fields:d}:
-            raise TypeError('Expected {num_fields:d} arguments, got %d' % len(result))
+        n = len(cls._fields)
+        if len(result) != n:
+            raise TypeError('Expected %d arguments, got %d' % (n, len(result)))
         return result
 
     def __repr__(self):
         'Return a nicely formatted representation string'
-        return '{typename}({repr_fmt})' % self
+        values = (getattr(self, name) for name in self._fields)
+        attrstring = ', '.join('%s=%r' % (k, v) for k, v in zip(self._fields, values))
```

Comment:
Slight less verbose: `', '.join('%s=%r' % kv for kv in zip(self._fields, values))`